### PR TITLE
Refactor metal3 components

### PR DESCRIPTION
**Pull request by jtomasek**
* Move Bare Metal Hosts components into a separate directory
  to make space for Bare Metal Nodes
* Refactor components to use default exports and use CamelCase naming convention
* Use BareMetalHost instead of BaremetalHost
* extracted host selectors into a separate module
* rename NodeCell and MachineCell to NodeLink and MachineLink
* other small fixes

### CODE REVIEW

File: frontend/packages/metal3-plugin/src/selectors/baremetal-hosts.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import { K8sResourceKind, MachineKind } from '@console/internal/module/k8s';
+import { BareMetalHostDisk } from '../types';
+import {
+  HOST_POWER_STATUS_POWERED_ON,
+  HOST_POWER_STATUS_POWERING_OFF,
+  HOST_POWER_STATUS_POWERING_ON,
+  HOST_POWER_STATUS_POWERED_OFF,
+} from '../constants';
+
+export const getHostOperationalStatus = (host: K8sResourceKind) =>
```
Comment (rawagner) [inline frontend/packages/metal3-plugin/src/selectors/baremetal-hosts.ts]: can you also add types for all of these selectors ?

Comment (jtomasek) [reply frontend/packages/metal3-plugin/src/selectors/baremetal-hosts.ts]: Done

**Pull request merged**